From 48c7bbd4d54b5182c34f1dc18ef95a4e34fba8fe Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 9 Feb 2019 10:30:23 +0530
Subject: [PATCH 0971/1051] octeontx2-af: Skip creating limits pci file for
 undiscovered RVU PFs

If not all of 15 RVU PFs (except AF) are enumerated in kernel then
current logic results in a kernel crash. This patch fixes it by
skipping 'limits' pci file for undiscovered PFs.

Change-Id: I399fafe4007625a3b2a44ca920ed2fcb436b65a2
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3727
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
index 0128eaa070ff..04573e110d4e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
@@ -525,6 +525,8 @@ static int rvu_create_limits_sysfs(struct rvu *rvu)
 
 	for (i = 0; i < rvu->hw->total_pfs; i++) {
 		pf = &rvu->pf[i];
+		if (!pf->pdev)
+			continue;
 		pdev = pf->pdev;
 
 		pf->limits_kobj = kobject_create_and_add("limits",
-- 
2.17.1

